core/src/main/java/hudson/util/StreamTaskListener.java

```diff
@@ -169,7 +169,7 @@ private void writeObject(ObjectOutputStream out) throws IOException {
     }
 
     private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
-        out = new PrintStream((OutputStream)in.readObject(),true);
+        out = new PrintStream((OutputStream)in.readObject(), false);
```

Comment:
> We expect that code flushes a stream when it is done with it

That expectation seems reasonable and preferable to the current behavior, but given that this code has been unmodified since 2007 I am a little concerned that this may break things. Should be easy enough to fix if it does cause issues, but I do not have a good sense of whether there may be widespread impact.

Reply:
> given that this code has been unmodified since 2007 I am a little concerned that this may break things

Indeed. Of course that is true of a lot of Jenkins code.